docs/docs/typechecking-with-proptypes.md

```diff
@@ -0,0 +1,157 @@
+---
+id: typechecking-with-proptypes
+title: Typechecking With PropTypes
+permalink: docs/typechecking-with-proptypes.html
+---
+
+As your app grows, you can catch a lot of bugs with typechecking. For some applications, you can use JavaScript extensions like [Flow](https://flowtype.org/) or [TypeScript](https://www.typescriptlang.org/) to typecheck your whole application. But even if you don't use those, React has some built-in typechecking abilities. To run typechecking on the props for a component, you can assign the special `propTypes` property:
+
+```javascript
+class Greeting extends React.Component {
+  render() {
+    return (
+      <h1>Hello, {this.props.name}</h1>
+    );
+  }
+}
+
+Greeting.propTypes = {
+  name: React.PropTypes.string
+};
+```
+
+`React.PropTypes` exports a range of validators that can be used to make sure the data you receive is valid. In this example, we're using `React.PropTypes.string`. When an invalid value is provided for a prop, a warning will be shown in the JavaScript console. For performance reasons, `PropTypes` is only checked in development mode.
+
+### React.PropTypes
+
+Here is an example documenting the different validators provided:
```

Comment:
This is a good list of supported validators. Do you think I should update /react/docs/react-api.html#proptypes to include a list of these? An API reference should list all valid symbols. This document can then cover use cases alongside code examples.